### PR TITLE
e2e workload tests on new storage class with replica and compression enabled rbd pool

### PR DESCRIPTION
**Pull request by shreekarSS**

### CODE REVIEW

File: tests/manage/storageclass/test_new_sc_rbd_e2e_workloads.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import logging
+import pytest
+
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import E2ETest, tier2
+
+
```
Comment (hnallurv) [inline tests/manage/storageclass/test_new_sc_rbd_e2e_workloads.py:7]: Looks like this test covers only the test https://polarion.engineering.redhat.com/polarion/#/project/OpenShiftContainerStorage/workitem?id=OCS-2345.

where do you cover the following tests?
https://polarion.engineering.redhat.com/polarion/#/project/OpenShiftContainerStorage/workitem?id=OCS-2347
https://polarion.engineering.redhat.com/polarion/#/project/OpenShiftContainerStorage/workitem?id=OCS-2346
https://polarion.engineering.redhat.com/polarion/#/project/OpenShiftContainerStorage/workitem?id=OCS-2344

Comment (shreekarSS) [reply tests/manage/storageclass/test_new_sc_rbd_e2e_workloads.py:7]: I'll update the parameters and test case IDs

File: tests/manage/storageclass/test_new_sc_rbd_e2e_workloads.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+@tier2
```
Comment (hnallurv) [inline tests/manage/storageclass/test_new_sc_rbd_e2e_workloads.py:8]: Please implement code to provide a comparison between with and without compression for each of the workloads

File: tests/manage/storageclass/test_new_sc_rbd_e2e_workloads.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+class TestCreateNewScWithNeWRbDPoolE2EWorkloads(E2ETest):
+
+    @pytest.mark.parametrize(
+        argnames=["replica", "compression"],
+        argvalues=[
+            pytest.param(
+                *[2, 'aggressive'], marks=pytest.mark.polarion_id("OCS-XXX")
```
Comment (PrasadDesala) [inline tests/manage/storageclass/test_new_sc_rbd_e2e_workloads.py:15]: Please add polarion IDs

File: tests/manage/storageclass/test_new_sc_rbd_e2e_workloads.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            )
+        ],
+    )
+    def test_new_sc_new_rbd_pool_amq(
+            self,
+            storageclass_factory,
+            amq_factory_fixture,
+            replica,
+            compression):
```
Comment (PrasadDesala) [inline tests/manage/storageclass/test_new_sc_rbd_e2e_workloads.py:24]: break the line after compression

File: tests/manage/storageclass/test_new_sc_rbd_e2e_workloads.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        interface_type = constants.CEPHBLOCKPOOL
+        sc_obj = storageclass_factory(
+            interface=interface_type,
+            new_rbd_pool=True,
+            replica=replica,
+            compression=compression,
+        )
+        self.amq, self.threads = amq_factory_fixture(sc_name=sc_obj.name)
+
```
Comment (PrasadDesala) [inline tests/manage/storageclass/test_new_sc_rbd_e2e_workloads.py:33]: delete empty line

File: tests/manage/storageclass/test_new_sc_rbd_e2e_workloads.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
+import logging
+import pytest
+
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import E2ETest, tier2
+
+
+@tier2
+class TestCreateNewScWithNeWRbDPoolE2EWorkloads(E2ETest):
```
Comment (PrasadDesala) [inline tests/manage/storageclass/test_new_sc_rbd_e2e_workloads.py:9]: does E2Eworkloads should be replaced with amq in the class name?

**Comment by shreekarSS** [comment]
Verified locally

**Comment by shreekarSS** [comment]
[test_logs.zip](https://github.com/red-hat-storage/ocs-ci/files/5689779/test_logs.zip)

File: ocs_ci/ocs/amq.py [hunk 7]
```diff
@@ -928,14 +929,23 @@ def cleanup(
...
             )
+
+            ocs_pvc_obj = get_all_pvc_objs(namespace=kafka_namespace)
+
         run_cmd(f"oc delete project {kafka_namespace}")
 
+        self.ns_obj.wait_for_delete(resource_name=kafka_namespace, timeout=90)
+        for pvc in ocs_pvc_obj:
+            logging.info(pvc.name)
+            from ocs_ci.helpers import helpers
```
Comment (ebenahar) [inline ocs_ci/ocs/amq.py:947]: please move the import to be outside the for loop

**Comment by shreekarSS** [comment]
> Unknown PR validation
> 
> Cluster Name:
> Cluster Configuration:
> PR Test Suite: tier2
> PR Test Path: tests/e2e/workloads/test_new_sc_rbd_e2e_workloads.py
> Additional Test Params:
> OCP VERSION: 4.6
> OCS VERSION: 4.6
> 
> [Job](https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-trigger-test-pr/3655/) state: ABORTED.

Pgsql failure is passed in the next test case.08:40:48 - MainThread - ocs_ci.ocs.pgsql - INFO - PGBench on pgbench-1-dbs-client-1-c970b57c-9tn44 completed successfully

**Review by Akarsha-rai — DISMISSED** [review]
LGTM

File: tests/e2e/workloads/test_new_sc_rbd_e2e_workloads.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+        self.cb = couchbase_factory_fixture(sc_name=sc_obj.name, run_in_bg=True)
+        self.pgsql = pgsql_factory_fixture(
+            replicas=3, clients=3, transactions=600, sc_name=sc_obj.name
+        )
+
+        bg_handler = flowtest.BackgroundOps()
+        bg_ops = [self.cb.result]
+        bg_handler.wait_for_bg_operations(bg_ops, timeout=3600)
+        cluster_used_space = get_percent_used_capacity()
+        log.info(
```
Comment (PrasadDesala) [inline tests/e2e/workloads/test_new_sc_rbd_e2e_workloads.py:53]: We may need to also validate that data is compressed

File: tests/e2e/workloads/test_new_sc_rbd_e2e_workloads.py [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+from ocs_ci.ocs.cluster import (
+    get_percent_used_capacity,
+)
+from ocs_ci.ocs import flowtest
+
+
+log = logging.getLogger(__name__)
+
+
+@ignore_leftovers
```
Comment (PrasadDesala) [inline tests/e2e/workloads/test_new_sc_rbd_e2e_workloads.py:15]: Any reason for ignoring leftovers here?

File: tests/e2e/workloads/test_new_sc_rbd_e2e_workloads.py [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+    )
+    def test_new_sc_new_rbd_pool_e2e_wl(
+        self,
+        storageclass_factory,
+        amq_factory_fixture,
+        couchbase_factory_fixture,
+        pgsql_factory_fixture,
+        replica,
+        compression,
+    ):
```
Comment (PrasadDesala) [inline tests/e2e/workloads/test_new_sc_rbd_e2e_workloads.py:35]: Please add description in comments on what this test script does

File: tests/e2e/workloads/test_new_sc_rbd_e2e_workloads.py (hunk 9 above)
Comment (shreekarSS) [reply tests/e2e/workloads/test_new_sc_rbd_e2e_workloads.py:15]: Taking  up in the follow-up pr

File: tests/e2e/workloads/test_new_sc_rbd_e2e_workloads.py (hunk 8 above)
Comment (shreekarSS) [reply tests/e2e/workloads/test_new_sc_rbd_e2e_workloads.py:53]: Taking up in the follow-up pr

**Review by Akarsha-rai — APPROVED** [review]

**Review by PrasadDesala — APPROVED** [review]

**Pull request merged**